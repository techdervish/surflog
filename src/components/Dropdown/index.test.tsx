import {
  expectWrappersTreeToMatchSnapshot,
  createShallowWrapper,
} from 'utils/testHelpers';
import Component from './index';

const createWrapper = createShallowWrapper(Component);

describe('Dropdown component', () => {
  const mockOnChange = jest.fn();

  const wrapper = createWrapper({
    options: ['1', '2', '3'],
    value: '3',
    onChange: mockOnChange,
  });

  it('renders the expected snapshot', () => {
    expectWrappersTreeToMatchSnapshot(wrapper);
  });

  it('should render pages and range dropdown', () => {
    expect(wrapper.find('Dropdown')).toHaveLength(1);
  });

  it('should call onChange on Dropdown change', () => {
    wrapper
      .find('Dropdown')
      .props()
      .onChange(1);
    expect(mockOnChange).toBeCalled();
  });
});
