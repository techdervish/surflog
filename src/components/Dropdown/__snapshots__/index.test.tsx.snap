// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component renders the expected snapshot 1`] = `
<Dropdown
  arrowClassName="source-dropdown-arrow"
  baseClassName="Dropdown"
  className="source-dropdown"
  controlClassName="source-dropdown-control"
  menuClassName="source-dropdown-menu"
  onChange={[MockFunction]}
  options={
    Array [
      "1",
      "2",
      "3",
    ]
  }
  value="3"
/>
`;
